geode-web-management/src/main/java/org/apache/geode/management/internal/rest/controllers/RegionManagementController.java

```diff
@@ -165,6 +167,37 @@ public ClusterManagementResult deleteRegion(
     Index filter = new Index();
     filter.setRegionPath(regionName);
     filter.setName(id);
-    return clusterManagementService.get(filter);
+    return addBiDirectionalUpLink(clusterManagementService.get(filter));
+  }
+
+  private static <R extends ClusterManagementGetResult> R addBiDirectionalDownLinks(R ret,
+      String... subs) {
+    ConfigurationResult<?, ?> result = ret.getResult();
+    addBiDirectionalDownLinks(result.getLinks(), subs);
+    return ret;
+  }
+
+  private static void addBiDirectionalDownLinks(Map<String, String> links, String... subs) {
+    for (String subCollection : subs) {
+      links.put(subCollection.replace("/", ""), links.get(SELF) + subCollection);
+    }
+  }
+
+  private static <R extends ClusterManagementGetResult> R addBiDirectionalUpLink(R ret) {
+    ConfigurationResult<?, ?> result = ret.getResult();
+    addBiDirectionalUpLink(result.getLinks());
+    return ret;
+  }
+
+  private static void addBiDirectionalUpLink(Map<String, String> links) {
+    String parent = links.get(SELF).replaceFirst("/[^/]*/[^/*]*$", "");
+    String type = parent.replaceFirst("s/[^/]*$", "").replaceFirst(".*/", "");
+    links.put(type, parent);
+  }
+
+  private static ClusterManagementListResult<Index, RuntimeInfo> addUplinks(
+      ClusterManagementListResult<Index, RuntimeInfo> list) {
+    list.getResult().forEach(r -> addBiDirectionalUpLink(r.getLinks()));
+    return list;
```

Comment:
these kind of operations somehow don't feel right. I thought it should be some interfaces that we just need to implement in order to add the links, not manipulate each and every end point. Have we explored other options to do this?

Reply:
do you want to rework this as part of this PR or a subsequent PR?

Reply:
what is not clear to me is where do we decide if we want to add the index link to the region result?

Reply:
PR was split so this is no longer relevant